internal/agent/device/fileio/writer.go

```diff
@@ -87,6 +88,35 @@ func (w *writer) MkdirAll(path string, perm os.FileMode) error {
 	return os.MkdirAll(filepath.Join(w.rootDir, path), perm)
 }
 
+func (w *writer) MkdirTemp(prefix string) (string, error) {
+	baseDir := w.rootDir
+	if baseDir == "" {
+		baseDir = os.TempDir()
+	}
```

Comment:
Isn't `w.rootDir` like a "chroot", i.e. wouldn't the temp dir be created "below" it?
```suggestion
	baseDir := filepath.Join(w.rootDir, os.TempDir())
```